---
title: "JB230212_ProjectNotebook"
output: html_document
date: "2023-02-12"
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# JB230210 Initialization of project.
* Reference genome:
  + hg38 patch 14.
  + https://www.ncbi.nlm.nih.gov/assembly/GCF_000001405.40
```{bash, eval=FALSE}
wget -O - https://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.40_GRCh38.p14/GCF_000001405.40_GRCh38.p14_genomic.fna.gz | gunzip -c > JB230210_hg38_patch14_genomic.fa
```
  
* Download the hg38 reference VCF from dbSNP
```{bash, eval=FALSE}
wget https://ftp.ncbi.nih.gov/snp/latest_release/VCF/GCF_000001405.40.gz
```

* Download ClinVAR variant summary tab delimited text file:

```{bash, eval=FALSE}
wget -O - https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/variant_summary.txt.gz | gunzip -c > JB230210_clinVAR_hg38_variants.txt
```

* Pre-requisites:
  + Bowtie2 index built from reference genome FASTA.

## Building bowtie2 index:
```{bash, eval=FALSE}

bowtie2-build -seed 111 /home/james/Desktop/Projects/GeneSys/External_Data/ncbi-genomes-2023-02-12/JB230210_hg38_patch14_genomic.fa /home/james/Desktop/Projects/GeneSys/External_Data/Bt2_hg38_idx/JB230210_hg38_bt2

```


## Obtaining pathogenic variants:
* There are multiple databases that exist for cataloging pathogenic variants.
  + Human Gene Mutation Database (HGMD): http://www.hgmd.org/
  + Online Mendelian Inheritance in Man (OMIM): https://omim.org/
  + Catalogue of Somatic Mutations in Cancer (COSMIC): http://cancer.sanger.ac.uk/cosmic
  + ClinVAR: https://www.ncbi.nlm.nih.gov/clinvar/
* Testing the ClinVAR example in this pipeline. 
  + Variant summary tab separated text file:
    + https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/variant_summary.txt.gz
  
## Downloading ClinVAR pathogenic variants summary:
* Tab separated text file:
  + Contains many columns but rsID will be the one to match called variants with. 

## Annotating called variants with rsIDs:
* Downloaded the hg38 reference VCF from dbSNP FTP server.
  + Plan to use bcftools-annotate to annotate rsID of called variants. 
  + Can then simply merge on matching rsID. 
 


# JB230220 Pipeline completes from start to end with a dry run successfully. 
* Fundamentals of the pipeline are in place for the basic test run. 
* Need to fully download the reference VCF. 
* Steps after this:
  + Full run to see if pipeline works as intended. 
  + Documentation of steps. 
  + Containerization via singularity. 
  + Add feature ideas to github page. + Update wiki with the documentation and DAG. 

  
#JB230224 Singularity container creation:
* To ensure reproducibility across machines I am configuring a singularity container that installs all required packages to run the snakemake command that executes the pipeline. 
* Commands can run with --fakeroot to enable building and running of containers without root privileges which is helpful when running on HPCCs.
* Pre-reqs: 
  + Install singularity. 
  + uidmap package must be installed on the machine to enable fakeroot. May have to ask server admin to do this for you with `sudo apt-get install -y uidmap`
  
```{bash, eval=F}

singularity build --fakeroot /home/james/Desktop/Projects/GeneSys/singularity_containers/genesys.sif  /home/james/Desktop/Projects/GeneSys/repos/GeneSys/singularity_genesys.def 

```

